build.gradle

```diff
@@ -182,3 +257,67 @@ diffCoverageReport {
         failOnViolation = true
     }
 }
+
+task integTest(type: RestIntegTestTask) {
+    description = "Run tests against a cluster"
+    testClassesDirs = sourceSets.test.output.classesDirs
+    classpath = sourceSets.test.runtimeClasspath
+}
+//tasks.named("check").configure { dependsOn(integTest) }
+
+task startTestExtension(type: Exec) {
+    commandLine 'bash', '-c', "./gradlew helloWorld &"
+}
+
+integTest {
+    dependsOn(startTestExtension)
+    systemProperty 'tests.security.manager', 'false'
+
+    systemProperty "https", System.getProperty("https")
+    systemProperty "user", System.getProperty("user")
+    systemProperty "password", System.getProperty("password")
+
+    // The 'doFirst' delays till execution time.
+    doFirst {
+        // Tell the test JVM if the cluster JVM is running under a debugger so that tests can
+        // use longer timeouts for requests.
+        def isDebuggingCluster = getDebug() || System.getProperty("test.debug") != null
+        systemProperty 'cluster.debug', isDebuggingCluster
+        // Set number of nodes system property to be used in tests
+        systemProperty 'cluster.number_of_nodes', "1"
+        // There seems to be an issue when running multi node run or integ tasks with unicast_hosts
+        // not being written, the waitForAllConditions ensures it's written
+        getClusters().forEach { cluster ->
+            cluster.waitForAllConditions()
+        }
+    }
+
+    // The --debug-jvm command-line option makes the cluster debuggable; this makes the tests debuggable
+    if (System.getProperty("test.debug") != null) {
+        jvmArgs '-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005'
+    }
+}
+
+task closeTestExtension (type: Exec) {
+    commandLine 'bash', '-c', "kill \$(lsof -i:4532)"
+}
+
+tasks.named("integTest").configure { finalizedBy(closeTestExtension) }
+
+testClusters.integTest {
+    extension(new ExtensionsProperties("hello-world", "opensearch-sdk-java-1", "127.0.0.1", "4532", "1.0", "3.0.0", "3.0.0"))
+    testDistribution = "ARCHIVE"
+    // Cluster shrink exception thrown if we try to set numberOfNodes to 1, so only apply if > 1
+    if (_numNodes > 1) numberOfNodes = _numNodes
```

Comment:
Will this condition ever work because we have defined `_numNodes = 1` above?

Reply:
Eventually, we will be pulling `_numNodes` from a SystemProperty and defaulting it to one.  This check will be necessary once that happens.

Reply:
We can set `_numNodes` like so : `def _numNodes = findProperty('numNodes') as Integer ?: 1`

Reply:
I changed the definition to read from the system property now